helm/nessie/templates/_helpers.tpl

```diff
@@ -92,3 +92,181 @@ names should coincide with jdbc schemes in connection URIs.
+{{- if .transport.maxRetryDelay -}}{{- $_ := set $map ( print $prefix "max-retry-delay" ) .transport.maxRetryDelay -}}{{- end -}}
+{{- if .transport.maxHttpConnections -}}{{- $_ := set $map ( print $prefix "max-http-connections" ) .transport.maxHttpConnections -}}{{- end -}}
+{{- if .transport.connectTimeout -}}{{- $_ := set $map ( print $prefix "connect-timeout" ) .transport.connectTimeout -}}{{- end -}}
+{{- if .transport.readTimeout -}}{{- $_ := set $map ( print $prefix "read-timeout" ) .transport.readTimeout -}}{{- end -}}
+{{- if .transport.writeTimeout -}}{{- $_ := set $map ( print $prefix "write-timeout" ) .transport.writeTimeout -}}{{- end -}}
+{{- if .transport.connectionIdleTimeout -}}{{- $_ := set $map ( print $prefix "connection-idle-timeout" ) .transport.connectionIdleTimeout -}}{{- end -}}
+{{- if .transport.readBlockSize -}}{{- $_ := set $map ( print $prefix "read-block-size" ) .transport.readBlockSize -}}{{- end -}}
+{{- if .transport.writeBlockSize -}}{{- $_ := set $map ( print $prefix "write-block-size" ) .transport.writeBlockSize -}}{{- end -}}
+{{- end -}}
+{{- list .advancedConfig ( print $prefix "configuration" ) $map | include "nessie.mergeAdvancedConfig" }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Define environkent variables for catalog storage options.
+*/}}
+{{- define "nessie.catalogStorageEnv" -}}
+{{- include "nessie.secretToEnv" (list .s3.accessKeySecret "awsAccessKeyId" "nessie.catalog.service.s3.access-key.name") }}
+{{- include "nessie.secretToEnv" (list .s3.accessKeySecret "awsSecretAccessKey" "nessie.catalog.service.s3.access-key.secret") }}
+{{- range $bucket := .s3.buckets -}}
+{{- include "nessie.secretToEnv" (list $bucket.accessKeySecret "awsAccessKeyId" (printf "nessie.catalog.service.s3.buckets.\"%s\".access-key.name" $bucket.name)) }}
+{{- include "nessie.secretToEnv" (list $bucket.accessKeySecret "awsSecretAccessKey" (printf "nessie.catalog.service.s3.buckets.\"%s\".access-key.secret" $bucket.name)) }}
+{{- end -}}
+{{- include "nessie.secretToEnv" (list .gcs.authCredentialsJsonSecret "key" "nessie.catalog.service.gcs.auth-credentials-json.key") }}
+{{- include "nessie.secretToEnv" (list .gcs.oauth2TokenSecret "token" "nessie.catalog.service.gcs.oauth-token.token") }}
+{{- include "nessie.secretToEnv" (list .gcs.oauth2TokenSecret "expiresAt" "nessie.catalog.service.gcs.oauth-token.expiresAt") }}
+{{- range $bucket := .gcs.buckets -}}
+{{- include "nessie.secretToEnv" (list $bucket.authCredentialsJsonSecret "key" (printf "nessie.catalog.service.gcs.buckets.\"%s\".auth-credentials-json.key" $bucket.name)) }}
+{{- include "nessie.secretToEnv" (list $bucket.oauth2TokenSecret "token" (printf "nessie.catalog.service.gcs.buckets.\"%s\".oauth-token.token" $bucket.name)) }}
+{{- include "nessie.secretToEnv" (list $bucket.oauth2TokenSecret "expiresAt" (printf "nessie.catalog.service.gcs.buckets.\"%s\".oauth-token.expiresAt" $bucket.name)) }}
```

Comment:
I have a doubt regarding this option: I need to check whether the key ends with `expiresAt` or whether it should be input as `expires-at`.

Reply:
Default naming would yield 'expires-at`

Reply:
You are right it's rendered `expires-at`.